racket/src/foreign/foreign.rktc

```diff
@@ -1914,8 +1914,17 @@ static void* SCHEME2C(const char *who,
   }
   res = scheme_malloc_fail_ok(mf,size);
   if (failok && (res == NULL)) scheme_signal_error("malloc: out of memory");
-  if (((from != NULL) || (foff != 0)) && (res != NULL))
-    memcpy(res, W_OFFSET(from, foff), size);
+
+  /* We might want to use foff as the object base address if from is NULL,
+   * therefore set a src point to use in memcpy to clarify this */
+  void *src = NULL;
```

Comment:
To accomodate the declaration and stick to C89 (which does matter for some of our supported build environments), wrap the new code with curly braces. Another possibility is to move the `src` declaration to the start of the function, but a local block seems better.

Reply:
Sure. Which build environments still require C89? Just out of curiosity...